src/js/modules/infragistics.ui.popover.js

```diff
@@ -162,38 +166,38 @@
 				this.options.selectors === undefined ) ?
 				this.element :
 				null;
-			this._priorityDir = [ "bottom", "top", "right", "left" ];
-			this._arrowDir = [ "top", "bottom", "left", "right" ];
+			this._arrowDir = {
+				"bottom": "top",
+				"top": "bottom",
+				"right": "left",
+				"left":"right"
+			};
 			this._positions = [ "balanced", "start", "end" ];
-			this._directionIndex = -1;
-			this._positionIndex = -1;
 			this._visible = false;
 			$( window ).on( "resize.popover", $.proxy( this._resizeHandler, this ) );
 		},
 		_createWidget: function (options, element) {
+			// initialization performance will be better if we don't try to normalize the defaults
```

Comment:
Options should not be cached in an internal property. If you're changing the value of the option, then assign it back to the option. This causes a lot of issues long term.